tests/unittests/HostManagerTest.cpp

```diff
@@ -130,27 +130,40 @@ TEST_F(HostManagerTest, runNetwork) {
   ready.wait();
   EXPECT_FALSE(errToBool(std::move(runErr)));
 }
-// This test is currently disabled, ASAN complains because the compiled function
-// can be freed out from under the DeviceManager. There are plans to moved
-// ownership of the compiledFunction to the deviceManager. Once that is done
-// this won't be an issue. Or we can add a callback to evictNetwork.
-
-// TEST_F(HostManagerTest, ConcurrentAddRemove) {
-//   constexpr auto numThreads = 6;
-//   constexpr auto numItersPerThread = 20;
-//   auto hostManager = createHostManager("CPU0", BackendKind::CPU);
-//   uint counter = 0;
-//   std::vector<std::thread> threads;
-//   for (auto i = 0; i < numThreads; ++i) {
-//     threads.emplace_back([&]() {
-//       for (auto j = 0; j < numItersPerThread; ++j) {
-//         addAndRemoveNetwork(hostManager.get(), counter);
-//         counter++;
-//       }
-//     });
-//   }
-
-//   for (auto &t : threads) {
-//     t.join();
-//   }
-// }
+
+TEST_F(HostManagerTest, ConcurrentAddRemoveUnique) {
+  constexpr auto numThreads = 6;
+  constexpr auto numItersPerThread = 20;
+  auto hostManager = createHostManager(BackendKind::CPU);
+  std::atomic<uint> counter{0};
+  std::vector<std::thread> threads;
+  for (auto i = 0; i < numThreads; ++i) {
+    threads.emplace_back([&]() {
+      for (auto j = 0; j < numItersPerThread; ++j) {
+        addAndRemoveNetwork(hostManager.get(), ++counter);
+      }
+    });
+  }
+
+  for (auto &t : threads) {
+    t.join();
+  }
+}
```

Comment:
what exactly is tested here? (no tsan/asan failures?)

Reply:
Yes, no TSAN/ASAN failures, and general stability.

Reply:
I'll add comments above the tests.